sdks/python/apache_beam/coders/row_coder.py

```diff
@@ -164,7 +164,13 @@ class RowCoderImpl(StreamCoderImpl):
   def __init__(self, schema, components):
     self.schema = schema
     self.constructor = named_tuple_from_schema(schema)
-    self.components = list(c.get_impl() for c in components)
+    self.encoding_positions = list(range(len(self.schema.fields)))
+    if any(field.encoding_position for field in self.schema.fields):
+      self.encoding_positions = list(
+          field.encoding_position for field in self.schema.fields)
```

Comment:
Sorry I just realized that the spec changed a bit from what I was thinking: https://github.com/apache/beam/blob/2e448dee58f1ee60551cc47b9aa7df6bc832734a/model/pipeline/src/main/proto/schema.proto#L52-L58

Note the part about `encoding_positions_set`. So this logic should look something like this (I also added a conservative check to verify encoding_positions are actually set):

```suggestion
    if self.schema.encoding_positions_set:
      if not all(field.encoding_position for field in self.schema.fields):
        raise ValueError(f"Schema with id {schema.id} has encoding_positions_set=True, but not all fields have encoding_position set")
      self.encoding_positions = list(
          field.encoding_position for field in self.schema.fields)
```

Reply:
Done.

Reply:
Please note the last part from the comment above:
> If this Field is part of a Schema where encoding_positions_set is True then encoding_position must be defined, *otherwise this field is ignored.*

What you have implemented now still respects the `encoding_position` fields even if  `encoding_positions_set` is False.